packages/api/src/routes/ehr/athenahealth/auth/middleware.ts

```diff
@@ -1,32 +1,31 @@
 import { NextFunction, Request, Response } from "express";
-import { getCxMapping } from "../../../../command/mapping/cx";
-import { getJwtToken } from "../../../../command/jwt-token";
 import { EhrSources } from "../../../../external/ehr/shared";
-import { getAuthorizationToken } from "../../../util";
+import {
+  processCxIdAsync,
+  processPatientRouteAsync,
+  processDocuemntRouteAsync,
+} from "../../shared";
+
+function parseAthenaHealthPracticeId(tokenData: { ah_practice?: string }): string {
+  const practiceId = tokenData.ah_practice;
+  if (!practiceId) throw new Error("Missing ah_practice on token data");
```

Comment:
I suggest we throw BadRequestError/4xx on these

Reply:
I converted all general Errors for parsing the request to BadRequest Errors.